Analysis/PWGDQ/include/PWGDQCore/VarManager.h

```diff
@@ -634,6 +636,29 @@ void VarManager::FillPair(T const& t1, T const& t2, float* values, PairCandidate
   values[kEta] = v12.Eta();
   values[kPhi] = v12.Phi();
   values[kRap] = -v12.Rapidity();
+  // CosTheta Helicity calculation
+  double px1 = t1.px();
+  double py1 = t1.py();
+  double pz1 = t1.pz();
+  double e1 = TMath::Sqrt(px1*px1 + py1*py1 + pz1*pz1 + m1*m1);
+  double px2 = t2.px();
+  double py2 = t2.py();
+  double pz2 = t2.pz();
+  double e2 = TMath::Sqrt(px2*px2 + py2*py2 + pz2*pz2 + m2*m2);
+  TLorentzVector pMuon1_Lab(px1,py1,pz1,e1);
+  TLorentzVector pMuon2_Lab(px2,py2,pz2,e2);
+  TLorentzVector pDimuon_Lab = pMuon1_Lab + pMuon2_Lab;
+  TVector3 beta = (-1./pDimuon_Lab.E())*pDimuon_Lab.Vect();
+  TLorentzVector pMuon1_CM = pMuon1_Lab;
+  TLorentzVector pMuon2_CM = pMuon2_Lab;
+  pMuon1_CM.Boost(beta);
+  pMuon2_CM.Boost(beta);
+  TVector3 zaxis;
+  zaxis = (pDimuon_Lab.Vect()).Unit();
+  double cosTheta;
+  if(t1.sign() > 0) cosTheta = zaxis.Dot((pMuon1_CM.Vect()).Unit());
+  else cosTheta = zaxis.Dot((pMuon2_CM.Vect()).Unit());
+  values[kCosThetaHE] = cosTheta;
```

Comment:
Can you use the newer version Math::LorentzVector instead of the legacy TLorentzVector? The newer class is faster and you can also initialize the 4-vectors using (px,py,pz,m), so no need to compute the energy.
For example, you could initialize the vectors like this:
Math::PxPyPzMVector pLeg1(t1.px(),t1.py(),t1.pz(),m1);

Try to use a different name than muon or electron if this is a general context. In this function, both dimuon and dielectron quantities are computed so maybe "lepton/dilepton" names can be used.